check_repos.sh

```diff
@@ -30,24 +32,29 @@ esac
 exit_status=0
 
 check_hooks_gitleaks() {
-    hooks_gitleak=$(cd $gitrepo; git config --bool hooks.gitleaks)
+    hooks_gitleak=$(cd "$gitrepo"; git config --bool hooks.gitleaks)
     if [ "$hooks_gitleak" = "true" ]; then
         return 0
     else
         return 1
     fi
 }
 
-check_precommit_hook() {
-    if [ -f $gitrepo/hooks/pre-commit ]; then
-      pcregrep -q '^((?!#).)*gitleaks\s+protect' $gitrepo/hooks/pre-commit
-      return $?
+check_hooks_path() {
+    # ensure that repos are not overriding the hookspath
+    hooks_path=$(cd "$gitrepo"; git config --local core.hooksPath)
+    if [ "$hooks_path" != "" ]; then
+        return 1
+    fi
+    hooks_path_origin=$(cd "$gitrepo"; git config --show-origin core.hooksPath)
+    if [[ "$hooks_path_origin" =~ /^file:$HOME/.gitconfig.*$HOME/.git-support/hooks ]]; then
+        return 1
```

Comment:
These two checks really verify the same thing. The first ensures that there is no local git config setting for `core.hookspath`. The second ensures that the `core.hookspath` is always pointing to the global git config in the $HOME directory and that it has the expected value. I suppose the second check is better and more complete?

Reply:
If they are duplicative, why not remove one? Sending you a message seperately